### PR TITLE
DRA: rename pkg/cm/dra/plugin files

### LINKED ISSUE #125924: DRA: rename pkg/cm/dra/plugin files
### What would you like to be added?

https://github.com/kubernetes/kubernetes/pull/125163#discussion_r1665552886:
- plugin.go -> registration.go
- client.go -> plugin.go

The PR above moved everything related to `Plugin` into `client.go` because that was where most of the code already was, but didn't renamed the files to avoid a larger diff and merge conflicts.

Once https://github.com/kubernetes/kubernetes/pull/125488 is merged, we can do the renaming.

### Why is this needed?

Code clarity.

/sig node

### PR DESCRIPTION
**Pull request by carlory**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Code clarity.

#### Which issue(s) this PR fixes:

Fixes #125924

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by pohly — CHANGES_REQUESTED** [review]
Please fix the linter error about `assert.NoError`, our linter is now a bit more strict than it was.

File: pkg/kubelet/cm/dra/plugin/plugin_test.go [hunk 1]
```diff
@@ -115,8 +115,9 @@ func TestGRPCConnIsReused(t *testing.T) {
 	m := sync.Mutex{}
 
 	p := &Plugin{
-		backgroundCtx: ctx,
-		endpoint:      addr,
+		backgroundCtx:     ctx,
+		endpoint:          addr,
+		clientCallTimeout: defaultClientCallTimeout,
```
Comment (carlory) [inline pkg/kubelet/cm/dra/plugin/plugin_test.go:120]: Fix https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/126504/pull-kubernetes-unit/1819208666302124032 after fixes lint

I'm not sure whether we should cherry-pick it? @pohly

Comment (bart0sh) [reply pkg/kubelet/cm/dra/plugin/plugin_test.go:120]: I don't think we need it as it's in testing code. However, I can do that if you decide otherwise.

Comment (carlory) [reply pkg/kubelet/cm/dra/plugin/plugin_test.go:120]: If it returns an error, we cannot determine whether it reuses the same gPRC connection from a testing perspective. 

https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/plugin/client_test.go#L151

**Comment by carlory** [comment]
replaced by https://github.com/kubernetes/kubernetes/pull/126834

**Pull request closed without merging**